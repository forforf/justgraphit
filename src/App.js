import React, { Component } from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { Router, Route, IndexRoute, hashHistory } from 'react-router';

import Container from './components/Container';
import Graphit from './components/Graphit/Graphit';
import About from './pages/About';

class App extends Component {
  render() {
    return (
      <MuiThemeProvider>
        <Router history={hashHistory}>
          <Route path="/" component={Container}>
            <IndexRoute component={Graphit}/>
            <Route path="/about" component={About}/>
          </Route>
        </Router>
      </MuiThemeProvider>
    );
  }
}

export default App;
