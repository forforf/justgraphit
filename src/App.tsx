import React from 'react';
import AppRouter from './components/AppRouter'
import {createTheme, responsiveFontSizes, ThemeProvider} from '@mui/material/styles';
import Storage from "./StorageModel/Storage";


// Create a theme instance.
let theme = createTheme();

theme = responsiveFontSizes(theme);

function App(): JSX.Element {

  // we want a single storage object per app instance.
  const initialObject = { "my first graph": [] };
  const storage = new Storage(initialObject);

  return (
    <ThemeProvider theme={theme}>
      <AppRouter storage={storage}/>
    </ThemeProvider>
  );
}

export default App;
