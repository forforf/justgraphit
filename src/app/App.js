
import React, { Component } from 'react';
import MuiThemeProvider from '../../node_modules/material-ui/styles/MuiThemeProvider';
import { Router, Route, IndexRoute, hashHistory } from 'react-router';

import Storage from 'StorageModel/Storage'

import Container from './header/Container';
import Graphit from './pages/Graphit/Graphit';
import About from './pages/About/About';


class App extends Component {

  constructor(props) {
    super(props);
    const storage = new Storage();

    // if storage is empty, initialize initial graph
    if (storage.isEmpty()) {
      storage.save("my first graph", []);
    }

    this.state = {
      storage: storage
    };
  }
  
  render() {
    const GraphItWrapper = (props) => {
      return <Graphit storage={this.state.storage}/>;
    };

    return (
      <MuiThemeProvider>
        <Router history={hashHistory}>
          <Route path="/" component={Container}>
            <IndexRoute component={GraphItWrapper}/>
            <Route path="/about" component={About}/>
          </Route>
        </Router>
      </MuiThemeProvider>
    );
  }
}

export default App;
