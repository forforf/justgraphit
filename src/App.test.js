import renderer from 'react-test-renderer';
import App from './App';

test('snapshot', () => {
  const component = renderer.create(
    <App />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
