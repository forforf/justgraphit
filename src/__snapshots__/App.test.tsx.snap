// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Array [
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
    title="JustGraphIt"
  />,
  <div
    className="MuiTabs-root css-1ujnqem-MuiTabs-root"
  >
    <div
      className="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
      onScroll={[Function]}
      style={
        Object {
          "marginBottom": -0,
          "overflow": "hidden",
        }
      }
    >
      <div
        aria-orientation={null}
        className="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
        onKeyDown={[Function]}
        role="tablist"
      >
        <a
          aria-selected={true}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          href="#/"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
        >
          GraphIt
          <span
            className="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
            style={Object {}}
          />
        </a>
        <a
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          href="#/edit"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
        >
          Editor
        </a>
        <a
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          href="#/about"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
        >
          About
        </a>
      </div>
    </div>
  </div>,
  <div
    className="Graphit"
  >
    <div
      className="wrapper"
    >
      <div
        className="row"
      >
        <div
          className="left col-2-3"
        >
          <div
            className="GraphInput"
          >
            <div
              className="GraphInput-name"
            >
              <h3>
                my first graph
              </h3>
            </div>
            <form>
              <div
                className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                onKeyPress={[Function]}
              >
                <div
                  className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    disabled={false}
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder="Name your graph"
                    required={false}
                    type="text"
                    value="my first graph"
                  />
                  <fieldset
                    aria-hidden={true}
                    className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="css-1a1ihhj"
                    >
                      <span
                        className="notranslate"
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
              </div>
              <br />
              <div
                className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                onKeyPress={[Function]}
              >
                <div
                  className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    disabled={false}
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder="Enter number to graph"
                    required={false}
                    type="number"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="css-1a1ihhj"
                    >
                      <span
                        className="notranslate"
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
              </div>
              <br />
              <button
                className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-sghohy-MuiButtonBase-root-MuiButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                Save
              </button>
            </form>
          </div>
          <div
            className="GraphChart"
          >
            <div>
              This is an EmptyGraph
            </div>
          </div>
        </div>
        <div
          className="right col-1-3"
        >
          <div
            className="GraphList"
          >
            <h3>
              Graph List
            </h3>
            <li
              className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
              disabled={false}
              onClick={[Function]}
            >
              <div
                className="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  my first graph
                </span>
              </div>
              <div
                className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="DeleteIcon"
                  focusable="false"
                  onClick={[Function]}
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </div>
            </li>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
